docs/user/save_load_resume_exp.md

```diff
@@ -0,0 +1,211 @@
+
+**Contents**:
+
+- [LocalRunner & LocalTFRunner for Garage experiment](#localrunner-localtfrunner-for-garage-experiment)
+- [Saving & Training Models in an experiment](#saving-training-models-in-an-experiment)
+- [Loading Models & Resuming an experiment](#loading-models-resuming-an-experiment)
+
+```python
+import garage
+from garage.experiment import LocalRunner, LocalTFRunner
+from garage.experiment.deterministic import set_seed
+```
+
+## LocalRunner & LocalTFRunner for Garage experiment
+
+`LocalRunner` class in Garage provides users with a range
+of utilities to set up environment and train an algorithm
+for an experiment.
+
+```Python
+runner = LocalRunner()
+```
+
+`LocalTFRunner` inherits `LocalRunner` class and provides
+a default TensorFlow session using Python context.
+
+```Python
+with LocalTFRunner(snapshot_config=ctxt) as runner:
+    ...
+```
+
+To perform the save, load and resume operations of
+an experiment, `Local_runner` provides four core functions:
+
+- `LocalRunner.save`: To save snapshot of a specific epoch.
+    This function uses [cloudpickle](https://github.com/cloudpipe/cloudpickle)
+    utility for serialization. All kind of Model, tensors and dictionaries
+    objects related to the experiment setup and training statistics
+    within the epoch with saved using this function.
+- `LocalRunner.setup`: To setup `LocalRunner` instance for
+    algorithm and environment in an experiment.
+- `LocalRunner.train`: To train an algorithm given the training
+    parameters.`
+- `LocalRunner.restore`: To restore an experiment from snapsnot.
+    This function uses cloudpickle's unplicking utilities to deserialize
+    pickled object files and re-setup the environment and retrieve model
+    data of a specified epoch.
+- `LocalRunner.resume`: To train an algorithm from a restored
+    experiment. This function provides the same interface as train().
+
+## Saving & Training Models in an experiment
+
+In general, saving and training models in an experiment includes
+ the following steps:
+
+- Initialize the `LocalRunner`/ `LocalTFRunner` instance
+- Define the environment and algorithms for an experiment
+- Setup the runner for algorithm and environment with `LocalRunner.setup`.
+- Run the training step with `LocalRunner.train`
+
+```Python
+runner = LocalRunner()
+env = Env(...)
+policy = Policy(...)
+algo = Algo(
+        env=env,
+        policy=policy,
+        ...)
+runner.setup(algo, env)
+runner.train(n_epochs=100, batch_size=4000)
+```
+
+## Loading Models & Resuming an experiment
+
+In general, loading models and resuming an experiment includes
+ the following steps:
+
+- Initialize the `LocalRunner`/ `LocalTFRunner` instance
+- Restore the algorithm and experiment setup with `LocalRunner.restore`
+- Define for the parameters we want to update during training
+- Run the training step with `LocalRunner.resume`
+
+```Python
+# to resume immediately.
+runner = LocalRunner()
+runner.restore(resume_from_dir)
+runner.resume
```

Comment:
bracket?

Reply:
Thanks for the spotting. Opened #1692 to fix that.